### ISSUE TITLE
E0122 "trait bounds are not (yet) enforced in type definitions" not triggered by `where` clauses

### ISSUE THREAD

**Issue by insaneinside** [rust-lang/rust#39122]
As [this playpen shows](https://is.gd/JS56xw), diagnostic E0122 does not catch trait bounds introduced via `where` clauses.

Specifically,
```rust
type X<T: std::ops::Add> = T;
```
triggers a warning, but

```rust
type Y<T> where T: std::ops::Add = T;
```
does _not_.

**Comment by brson** [comment]
Looks like something that should be locked down.

**Comment by circuitfox** [comment]
I think I'll take a shot at this.
